packages/static-config/src/swc.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  ArrayExpression,
+  BooleanLiteral,
+  ExportDeclaration,
+  Identifier,
+  KeyValueProperty,
+  Module,
+  Node,
+  NullLiteral,
+  NumericLiteral,
+  ObjectExpression,
+  StringLiteral,
+  VariableDeclaration,
+} from '@swc/core';
+import { FromSchema, JSONSchema } from 'json-schema-to-ts';
+import { validate } from './validation';
+
+function isExportDeclaration(node: Node): node is ExportDeclaration {
+  return node.type === 'ExportDeclaration';
+}
+
+function isVariableDeclaration(node: Node): node is VariableDeclaration {
+  return node.type === 'VariableDeclaration';
+}
+
+function isIdentifier(node: Node): node is Identifier {
+  return node.type === 'Identifier';
+}
+
+function isBooleanLiteral(node: Node): node is BooleanLiteral {
+  return node.type === 'BooleanLiteral';
+}
+
+function isNullLiteral(node: Node): node is NullLiteral {
+  return node.type === 'NullLiteral';
+}
+
+function isStringLiteral(node: Node): node is StringLiteral {
+  return node.type === 'StringLiteral';
+}
+
+function isNumericLiteral(node: Node): node is NumericLiteral {
+  return node.type === 'NumericLiteral';
+}
+
+function isArrayExpression(node: Node): node is ArrayExpression {
+  return node.type === 'ArrayExpression';
+}
+
+function isObjectExpression(node: Node): node is ObjectExpression {
+  return node.type === 'ObjectExpression';
+}
+
+function isKeyValueProperty(node: Node): node is KeyValueProperty {
+  return node.type === 'KeyValueProperty';
+}
+
+export class UnsupportedValueError extends Error {}
+
+function extractValue(node: Node): any {
```

Comment:
Why is the return type `any` if going through all the trouble of getting the types correct?

Reply:
Because it returns arbitrary literals but I agree that we can do better. To avoid circular references of `Value`, I defined as follows in the latest commit:

```ts
export type Value = undefined | null | boolean | string | number | any[] | Record<string, any>;
```